MIGRATION.md

```diff
@@ -0,0 +1,245 @@
+
+Response response = db.update(p); // the same object is used for the update, it will set the name parameter to new_name
+```
+
+#### 1. Use the `Document` model class, storing user properties in the `Map`
+
+```java
+//set up the service client
+GetDocumentOptions documentOptions =
+        new GetDocumentOptions.Builder()
+                .db("example_db")
+                .docId("example_id")
+                .build();
+
+Document doc = service.getDocument(documentOptions)
+        .execute()
+        .getResult();
+
+System.out.println(doc); // will be a JSON
+
+// Set the JSON properties to the Pojo
+Pojo p = new Pojo();
+p.setName((String) doc.getProperties().get("name"));
+System.out.println(p); // will be a Pojo with the same name as the JSON
+
+p.setName("new_name");
+System.out.println(p.getName()); // new_name
+
+doc.put("name", p.getName()); // add your modifications to the Document object
+
+PutDocumentOptions putDocumentOptions =
+        new PutDocumentOptions.Builder()
+                .db("example_db")
+                .docId("example_id")
+                .document(doc)
+                .build();
+
+DocumentResult response =
+        service.putDocument(putDocumentOptions)
+                .execute()
+                .getResult();
+```
+
+#### 2. Convert the `Document` model into a POJO (and vice versa)
+
+```java
+//set up the service client
+GetDocumentOptions documentOptions =
+        new GetDocumentOptions.Builder()
+                .db("example_db")
+                .docId("example_id")
+                .build();
+
+Document doc = service.getDocument(documentOptions)
+        .execute()
+        .getResult();
+
+System.out.println(doc); // will be a JSON
+
+// Serialize the JSON to Pojo
+Pojo p = YourJsonSerializer.fromJson(doc.toString(), Pojo.class);
+System.out.println(p); // the value of the Pojo's toString method
+
+p.setName("new_name");
+System.out.println(p.getName()); // will be new_name
+
+// Deserialize the Pojo back to the Document model
+doc.setProperties(YourJsonSerializer.fromJson(YourJsonSerializer.toJson(p), Map.class));
+
+PutDocumentOptions putDocumentOptions =
+                new PutDocumentOptions.Builder()
+                        .db("example_db")
+                        .docId("example_id")
+                        .document(doc)
+                        .build();
+
+DocumentResult response =
+        service.putDocument(putDocumentOptions)
+                .execute()
+                .getResult();
+```
+
+#### 3. Bypass the `Document` model and use the `AsStream` methods
+
+```java
+//set up the service client
+GetDocumentOptions documentOptions =
+        new GetDocumentOptions.Builder()
+                .db("example_db")
+                .docId("example_id")
+                .build();
+
+Pojo p = new Pojo()
+try(InputStream is = service.getDocumentAsStream(documentOptions).execute().getResult()){
+    InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8);
```

Comment:
I'd just drop this line, handling of a raw `InputStream`/`Reader` will be serializer dependent anyway - and we're missing the equivalent charset on the `getBytes()` for the put.
```suggestion
```